Implements EventClass::Execute cases, and functions used by those cases.

tomsons26 opened the pull request.

tomsons26 committed:
Implements most EventClass::Execute cases, and functions used by those cases.

tomsons26 commented on src/game/common/globals.cpp:
```diff
@@ -172,6 +176,9 @@ void *g_WakeShapes = nullptr;
 void *g_TurretShapes = nullptr;
 void *g_SamShapes = nullptr;
 void *g_MGunShapes = nullptr;
+int g_FormMove = 0;
```
not sure about the default values here, there are none in original

tomsons26 commented on src/game/engine/house.cpp:
```diff
@@ -823,6 +824,93 @@ BOOL HouseClass::Is_Ally(ObjectClass *object) const
     return Is_Ally(object->Owner());
 }
 
+ProdFailType HouseClass::Begin_Production(RTTIType rtti, int id)
```
not sure on the position in the header and cpp for these functions, any specific order they should follow?

tomsons26 commented on src/game/engine/house.h:
```diff
@@ -55,6 +55,14 @@ struct ZoneInfo
     int Infantry;
 };
 
+enum ProdFailType
```
Anything known about this?

tomsons26 committed:
Implements most EventClass::Execute cases, and functions used by those cases.

OmniBlade commented on src/game/engine/building.h:
```diff
@@ -36,6 +36,16 @@ class BuildingClass : public TechnoClass
+    virtual void Enter_Idle_Mode(BOOL a1 = false) override;
+    virtual void Grand_Opening(int a1);
+    virtual void Update_Buildables();
+    virtual void * Remap_Table();
```
Formatting issue with placement of * for the pointer, clang-format will fix it.

OmniBlade commented on src/game/engine/foot.cpp:
```diff
@@ -1255,3 +1256,31 @@ PathType *FootClass::Find_Path_Wrapper(cell_t dest, FacingType *buffer, int leng
+void FootClass::Queue_Navigation_List(target_t target)
+{
+    int i;
+    if (target) {
```
Code style, you should be explict with conditionals, here it should be `if (target != 0)`.

OmniBlade commented on src/game/engine/foot.cpp:
```diff
@@ -1255,3 +1256,31 @@ PathType *FootClass::Find_Path_Wrapper(cell_t dest, FacingType *buffer, int leng
+{
+    int i;
+    if (target) {
+        for (i = 0; i < NAV_LENGTH && m_NavList[i]; ++i);
```
And again here, `m_NavList[i] != 0`.

OmniBlade commented on src/game/engine/foot.cpp:
```diff
@@ -1255,3 +1256,31 @@ PathType *FootClass::Find_Path_Wrapper(cell_t dest, FacingType *buffer, int leng
+    if (target) {
+        for (i = 0; i < NAV_LENGTH && m_NavList[i]; ++i);
+        if (target != As_Target() || i <= 0) {
+            if (!i) {
```
And again here, `i == 0`.

OmniBlade commented on src/game/engine/gameevent.h:
```diff
@@ -81,14 +82,14 @@ class GameEventClass
     GameEventClass(GameEventType type, RTTIType rtti, cell_t cell); // For events:
     GameEventClass(GameEventType type, SpecialWeaponType special, cell_t cell); // For events: SPECIAL_PLACE
-    GameEventClass(GameEventType type, void *a2, unsigned long a3); // For events:
+    GameEventClass(GameEventType type, void *a2, unsigned long a3); // For events: ADDPLAYER
```
Don't use types that don't have a specific size where a specific size is actually required. Expecially don't use long and unsigned long as they are known to have different sizes on different platforms and architectures. Use uint32_t in place of unsigned long, uint16_t in place of unsigned short and uint8_t in place of unsigned char.

OmniBlade commented on src/game/engine/gameevent.h:
```diff
@@ -132,93 +135,113 @@ class GameEventClass
         unsigned char m_Delay;
     };
 
+    // confirmed - TIMING
```
Are we sure this is a bitfield, looks like two uint16_t's if the bitfield has them as 16 bits?

OmniBlade commented on src/game/engine/gameevent.cpp:
```diff
@@ -348,23 +307,35 @@ GameEventClass::GameEventClass(AnimType anim, HousesType owner, coord_t coord) :
 
     switch (m_Type) {
-        case EVENT_ALLY:
+        // done
```
You don't need to leave comments saying done in code intended to be final, just leave todos on stuff needing doing and everything else is assumed to be done.

tomsons26 committed:
Fixes issues in some events, fixes Session struct.
Adds a method in watcom to check a specific offset and documents a undocumented pragma for dumping compiled type information
Adds SessionClass to sizeof checks
